src/web/app/components/teammates-common/submission-status-tooltip.pipe.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { FeedbackSessionSubmissionStatus } from '../../../types/api-output';
+import { SubmissionStatusTooltipPipe } from './submission-status-tooltip.pipe';
+
+describe('SubmissionStatusTooltipPipe', () => {
+  it('create an instance', () => {
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transform with no deadlines correctly', () => {
+    jest.useFakeTimers().setSystemTime(new Date('2020-01-01').getTime());
+    const hasNoDeadlines = {
+      studentDeadlines: {},
+      instructorDeadlines: {},
+    };
+    const hasNoOngoingDeadlines = {
+      studentDeadlines: { nonOngoingExtension1: new Date('2019-01-01').valueOf() },
+      instructorDeadlines: { nonOngoingExtension2: new Date('2019-02-01').valueOf() },
+    };
+    const pipe: SubmissionStatusTooltipPipe = new SubmissionStatusTooltipPipe();
+
+    expect(pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN, hasNoDeadlines)).toEqual(
+      pipe.transform(FeedbackSessionSubmissionStatus.VISIBLE_NOT_OPEN),
+    );
```

Comment:
These should probably expect the output to be equal to a hardcoded string. The tests here should not just test that the behavior with and without extensions are equal, I don't think this test really makes sense without existing tests for the default behavior as well.

Reply:
Makes sense, I've added another test for the default behavior too

Reply:
![image](https://user-images.githubusercontent.com/10182564/162865850-eb26fbc5-8a8d-4276-9095-c0fcfa0f4df6.png)

But testing for the default behavior through hardcoding does seem very redundant too. I'm thinking that the this test should be fine in serving it's purpose, but I'm open to suggestions

Reply:
Personally think it's fine.